testsuite/features/support/commonlib.rb

```diff
@@ -48,6 +48,14 @@ def compute_channels_to_leave_running
   do_not_kill.uniq
 end
 
+# This function returns the correct channel names defined in 'constants.rb' depending on the product
+# Params:
+# +channel_name:: the channel you want to get.
+# See 'constants.rb' for more information on what channel constants we have
+def get_product_channel(channel_name)
+  channel_name[product]
```

Comment:
I would maybe use the word "channels_map" as what you provide as parameter into the method is directly the constant of the map. Not a channel name in fact.

Reply:
A suggestion could be to have:

```suggestion
def get_os_channel_from_map(channel_map, map_key)
  channel_name[product][map_key]
end  
```

Reply:
Another approach, could be to keep using just LABEL_BY_BASE_CHANNEL and BASE_CHANNEL_BY_CLIENT, being transparent from the rest of the code, and zero changes.
And when we initialize these constants, we keep into account the value passed by environment variable, we use again the strategy of compare `ENV['SERVER'].include? 'uyuni'` and append some entries to the map depending on this comparison.

Reply:
I think your last suggestions is more complicated since in Uyuni we not only append `-uyuni` to a channel, but also have completely different channel names. 

Example:
```ruby
LABEL_BY_BASE_CHANNEL = { 'SUSE Manager' => {
                            'almalinux9 for x86_64' => 'no-appstream-alma-9-result-almalinux9-x86_64',
                            'RHEL x86_64 Server 7' => 'rhel-x86_64-server-7',
                            'oraclelinux9 for x86_64' => 'no-appstream-oracle-9-result-oraclelinux9-x86_64',
                            'RHEL8-Pool for x86_64' => 'no-appstream-8-result-rhel8-pool-x86_64',
                            'rockylinux-9 for x86_64' => 'no-appstream-9-result-rockylinux-9-x86_64',
                            'ubuntu-18.04-pool' => 'ubuntu-18.04-pool-amd64',
                            'ubuntu-2004-amd64-main' => 'ubuntu-2004-amd64-main-amd64',
                            'ubuntu-2204-amd64-main' => 'ubuntu-2204-amd64-main-amd64',
                          },
                          'Uyuni' => {
                            'almalinux9 for x86_64' => 'almalinux9-x86_64',
                            'Fake Base Channel' => 'fake_base_channel',
                            'RHEL x86_64 Server 7' => 'centos7-x86_64',
                            'oraclelinux9 for x86_64' => 'oraclelinux9-x86_64',
                            'RHEL8-Pool for x86_64' => 'rockylinux8-x86_64',
                            'rockylinux-9 for x86_64' => 'rockylinux9-x86_64',
                            'ubuntu-18.04-pool' => 'ubuntu-18.04-pool-amd64-uyuni',
                            'ubuntu-2004-amd64-main' => 'ubuntu-20.04-pool-amd64-uyuni',
                            'ubuntu-2204-amd64-main' => 'ubuntu-22.04-pool-amd64-uyuni',
                          } }.freeze

```